lib/pact.ex

```diff
@@ -25,8 +25,39 @@ defmodule Pact do
 
   use GenServer
 
+  @doc """
+    Replace the given `name` for `pid` with the given expression. This will
+    generate a new module with the given methods.
+
+    ## Example
+
+    ```
+    import Pact
+
+    Pact.put(:enum, Enum)
+    Pact.replace self, :enum do
+      def map(_map, _fn) do
+        [1, 2, 3]
+      end
+    end
+    ```
+
+    So now if you call `Pact.get(:enum).map(%{}, fn -> end)` it will return
+    `[1, 2, 3]`.
+  """
+  defmacro replace(pid, name, expression) do
+    body = Keyword.get(expression, :do, nil)
+    uid = :base64.encode(:crypto.strong_rand_bytes(15))
```

Comment:
I like this. Does it need to be that long though? That will make errors weird having an extra 15 character long module name. Maybe just 5 is fine?